### ISSUE TITLE
Rule proposal: prefer `obj.entries()` to create `URLSearchParams`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2269]
### Description

It might be a bit niche, but not everyone knows that you can convert an object to a URLSearchParams without a for loop.

### Fail

```js
const tokenParams = new URLSearchParams();
for (const [param, value] of Object.entries(tokenBody)) {
  tokenParams.set(param, value);
}
```

### Pass

```js
new URLSearchParams(Object.entries(tokenBody))
```

```js
// This piece of code is adding entries to an existing `URLSearchParam`
for (const [param, value] of Object.entries(tokenBody)) {
  tokenParams.set(param, value);
}
```

**Comment by silverwind** [comment]
Do you even need the `Object.entries`?

```js
> new URLSearchParams(Object.entries({a: 1}))
URLSearchParams { 'a' => '1' }
> new URLSearchParams({a: 1})
URLSearchParams { 'a' => '1' }

**Comment by fregante** [comment]
Judging by MDN, that's seems to be accepted. I think the problem used to be typescript, but that works now too

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams

**Comment by silverwind** [comment]
I guess this rule could apply to more than just `URLSearchParams`, basically anything that accepts a iterable in its constructor. [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set) is such an example, [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData) is not.

**Comment by fregante** [comment]
Objects are not iterable

**Comment by silverwind** [comment]
Right, so `URLSearchParams` is special in that it accepts them, but the basic idea is the same in all cases, avoid a loop when a constructor can do the looping itself.

**Comment by sindresorhus** [comment]
Accepted. `prefer-iterable-in-constructor`
